### ISSUE TITLE
Create contribution guide and accessibility standards

### ISSUE THREAD

**Issue by benjaminwil** [simple-icons/simple-icons#472]
I was just reading up on making SVGs accessible and wonder if these SVGs should start including `<title>` tags for accessibility?

For example:

```
<svg viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg" fill-rule="evenodd" clip-rule="evenodd" stroke-linejoin="round" stroke-miterlimit="1.414">
  <title>Bandcamp</title>
  <path d="M0 12.5l4.958-9H16l-4.958 9H0z"/>
</svg>
```

[See the W3's Accessibility Features of SVG article](https://www.w3.org/TR/SVG-access/).

Right now only a few of the items include `<title>` tags, but they are generic (bad) titles generated by vector apps like Sketch.

If you'll consider this issue, I'm happy to write something up for the new `CONTRIBUTING.md`.

**Comment by danleech** [comment]
@benjaminwil Yes. 100% yes. Thanks for the link 😄

**Comment by danleech** [comment]
It’d be great to have a template for SVG files in `CONTRIBUTING.md`. Something like:

```svg
<svg aria-labelledby="title" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
    <title>[ICON_NAME] icon</title>
    <path … />
</svg>
```

> - Include the SVG namespace declaration `xmlns="http://www.w3.org/2000/svg"`. This is not required if the SVG is included inline on HTML5 web pages.
> - Include a `<title>` element for accessibility.
> - Add `aria-labelledby="title"` to the `<svg>` element to improve screen reader support.
> - Use a `viewBox` of 24×24, following Google’s [Material Design guidelines for system icons](https://material.io/guidelines/style/icons.html#icons-system-icons).

**Comment by benjaminwil** [comment]
I added a draft of the `CONTRIBUTING.md` in my pull request #475.

**Comment by benjaminwil** [comment]
@birjolaxew reviewed my pull request and suggested that `role="img"` also be added to the `<svg>` element for accessibility [and linked to this article on CSS-Tricks](https://css-tricks.com/accessible-svgs/). I'll add this to the `CONTRIBUTING.md` as well.

He also asked: 

> Almost all existing SVG's use a viewBox of 16 by 16. The only ones that don't are HTML5 (14x16) and Nucleo (24x24), both of which I'd say are simply bugs. Why the change?

The precedent for that change is in [Google’s Material Design guidelines for system icons](https://material.io/guidelines/style/icons.html#icons-system-icons), but @danleech – could you confirm that you'd like the standard viewbox for new icons to be 24 by 24 rather than 16 by 16?

**Comment by birjolaxew** [comment]
Just to expand a bit, Google's Material Design are for designing system icons, where it is useful to have a grid so you can keep consistent spacing. Simple Icons aren't designing anything, so this doesn't help us at all. The only reason for switching to 24x24 would be "because others are doing it", which is a fairly poor justification.

**Comment by benjaminwil** [comment]
Even if switching `viewBoxes` to 24 by 24 is justified, it'd be a major project on its own to refactor all of the existing icons.

I'd like to recommend a `viewBox` of 16 by 16 for now, and if it becomes possible to automate the re-sizing of the existing SVGs, we can update the contributing guidelines as well as all of the icons.

**Comment by danleech** [comment]
@benjaminwil You’re right it’s going to be a major project, but one I’m going to have a go at soon. The problem with the existing 16x16 icons is that they’re not proportional, so when icons are used side-by-side they don’t look harmonious.

**Comment by birjolaxew** [comment]
@danleech I'm not sure I follow on that last part. Could you give an example?

**Comment by danleech** [comment]
@birjolaxew That was poorly worded, sorry. Icons of different shapes should be _optically scaled_ so they look harmonious as a set. For example, 16px squares look bigger than 16px circles, so circles need to be made slightly larger.

Google’s system icon grid and key shapes is a great basis for icon scaling: https://material.io/guidelines/style/icons.html#icons-system-icons.

Here’s an example of five icons to illustrate the point. Icons on the top row are all 16x16px. Icons on the second row follow Google’s guidelines. The square icons are 18x18px, the round and irregular shapes are 20x20px. The Twitter icon is also shifted down by one pixel so it looks more inline.

![Proportional icons](https://lh4.googleusercontent.com/zLt95tIU_ukV2L7AmtCk58A0JXYffz-zRJFjYqp9SLjBJFAReQxRspCZKIsbxy70a6i23u-HRos97XI=w1576-h1307-rw)

**Comment by benjaminwil** [comment]
Thanks for the explanation @danleech.

I'm going to add a more verbose, plain speech description around how to size out icons depending on their shapes (rectangular, circular, or irregular) and add it to the `CONTRIBUTING.md`. I may also add that example of proportional icons.

After that, I'll close this issue. 👏

**Comment by gatzkerob** [comment]
`aria-labelledby="title"` is failing w3 validation:

"Error: The aria-labelledby attribute must point to an element in the same document"

You could add an `id="title"` to `<title>,` but if you use multiple icons on the same page you'll end up with:

"Error: Duplicate ID title"

I don't think there's a way get around this as of this post, aside from using ARIA 1.0 while omitting the `aria-labelledby` attribute.

Something to keep in mind.

More info here: [https://www.sitepoint.com/tips-accessible-svg/](https://www.sitepoint.com/tips-accessible-svg/)

**Comment by davidklebanoff** [comment]
This issue has been addressed by PR #475 and subsequently PR #640.

For the time being, it has been decided that all icons will have the same sized viewbox 24x24 and are scaled to fit the viewbox (Issue #582).

There is however, some effort into looking at optically sizing the icons in the future (Issue #477).
